### PR TITLE
Remove deprecated methods

### LINKED ISSUE #866: Remove deprecated code for 1.0.0 release
There are currently 124 functions that are intended to be deprecated by release 1.0.0. These functions should be removed before the 1.0.0 release.

```
❯ grep -rn "@utils.deprecated_method("
hvac/adapters.py:213:    @utils.deprecated_method(
hvac/v1/__init__.py:301:    @utils.deprecated_method(
hvac/v1/__init__.py:313:    @utils.deprecated_method(
hvac/v1/__init__.py:477:    @utils.deprecated_method(
hvac/v1/__init__.py:524:    @utils.deprecated_method(
hvac/v1/__init__.py:547:    @utils.deprecated_method(
hvac/v1/__init__.py:594:    @utils.deprecated_method(
hvac/v1/__init__.py:608:    @utils.deprecated_method(
hvac/v1/__init__.py:622:    @utils.deprecated_method(
hvac/v1/__init__.py:666:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:690:    @utils.deprecated_method(
hvac/v1/__init__.py:706:    @utils.deprecated_method(
hvac/v1/__init__.py:740:    @utils.deprecated_method(
hvac/v1/__init__.py:801:    @utils.deprecated_method(
hvac/v1/__init__.py:837:    @utils.deprecated_method(
hvac/v1/__init__.py:872:    @utils.deprecated_method(
hvac/v1/__init__.py:891:    @utils.deprecated_method(
hvac/v1/__init__.py:907:    @utils.deprecated_method(
hvac/v1/__init__.py:934:    @utils.deprecated_method(
hvac/v1/__init__.py:955:    @utils.deprecated_method(
hvac/v1/__init__.py:971:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:1009:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:1025:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:1038:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:1076:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:1092:    @utils.deprecated_method(to_be_removed_in_version="1.0.0")
hvac/v1/__init__.py:1105:    @utils.deprecated_method(
hvac/v1/__init__.py:1146:    @utils.deprecated_method(
hvac/v1/__init__.py:1160:    @utils.deprecated_method(
hvac/v1/__init__.py:1174:    @utils.deprecated_method(
hvac/v1/__init__.py:1198:    @utils.deprecated_method(
hvac/v1/__init__.py:1216:    @utils.deprecated_method(
hvac/v1/__init__.py:1233:    @utils.deprecated_method(
hvac/v1/__init__.py:1343:    @utils.deprecated_method(
hvac/v1/__init__.py:1359:    @utils.deprecated_method(
hvac/v1/__init__.py:1375:    @utils.deprecated_method(
hvac/v1/__init__.py:1394:    @utils.deprecated_method(
hvac/v1/__init__.py:1483:    @utils.deprecated_method(
hvac/v1/__init__.py:1504:    @utils.deprecated_method(
hvac/v1/__init__.py:1521:    @utils.deprecated_method(
hvac/v1/__init__.py:1536:    @utils.deprecated_method(
hvac/v1/__init__.py:1554:    @utils.deprecated_method(
hvac/v1/__init__.py:1575:    @utils.deprecated_method(
hvac/v1/__init__.py:1591:    @utils.deprecated_method(
hvac/v1/__init__.py:1632:    @utils.deprecated_method(
hvac/v1/__init__.py:1652:    @utils.deprecated_method(
hvac/v1/__init__.py:1671:    @utils.deprecated_method(
hvac/v1/__init__.py:1695:    @utils.deprecated_method(
hvac/v1/__init__.py:1715:    @utils.deprecated_method(
hvac/v1/__init__.py:1739:    @utils.deprecated_method(
hvac/v1/__init__.py:1765:    @utils.deprecated_method(
hvac/v1/__init__.py:1793:    @utils.deprecated_method(
hvac/v1/__init__.py:1836:    @utils.deprecated_method(
hvac/v1/__init__.py:1852:    @utils.deprecated_method(
hvac/v1/__init__.py:1915:    @utils.deprecated_method(
hvac/v1/__init__.py:1933:    @utils.deprecated_method(
hvac/v1/__init__.py:1949:    @utils.deprecated_method(
hvac/v1/__init__.py:1967:    @utils.deprecated_method(
hvac/v1/__init__.py:1990:    @utils.deprecated_method(
hvac/v1/__init__.py:2033:    @utils.deprecated_method(
hvac/v1/__init__.py:2050:    @utils.deprecated_method(
hvac/v1/__init__.py:2065:    @utils.deprecated_method(
hvac/v1/__init__.py:2082:    @utils.deprecated_method(
hvac/v1/__init__.py:2120:    @utils.deprecated_method(
hvac/v1/__init__.py:2137:    @utils.deprecated_method(
hvac/v1/__init__.py:2161:    @utils.deprecated_method(
hvac/v1/__init__.py:2217:    @utils.deprecated_method(
hvac/v1/__init__.py:2258:    @utils.deprecated_method(
hvac/v1/__init__.py:2304:    @utils.deprecated_method(
hvac/v1/__init__.py:2339:    @utils.deprecated_method(
hvac/v1/__init__.py:2368:    @utils.deprecated_method(
hvac/v1/__init__.py:2399:    @utils.deprecated_method(
hvac/v1/__init__.py:2431:    @utils.deprecated_method(
hvac/v1/__init__.py:2483:    @utils.deprecated_method(
hvac/v1/__init__.py:2540:    @utils.deprecated_method(
hvac/v1/__init__.py:2547:    @utils.deprecated_method(
hvac/v1/__init__.py:2555:    @utils.deprecated_method(
hvac/v1/__init__.py:2583:    @utils.deprecated_method(
hvac/v1/__init__.py:2590:    @utils.deprecated_method(
hvac/v1/__init__.py:2597:    @utils.deprecated_method(
hvac/v1/__init__.py:2604:    @utils.deprecated_method(
hvac/v1/__init__.py:2611:    @utils.deprecated_method(
hvac/v1/__init__.py:2618:    @utils.deprecated_method(
hvac/v1/__init__.py:2625:    @utils.deprecated_method(
hvac/v1/__init__.py:2632:    @utils.deprecated_method(
hvac/v1/__init__.py:2652:    @utils.deprecated_method(
hvac/v1/__init__.py:2682:    @utils.deprecated_method(
hvac/v1/__init__.py:2702:    @utils.deprecated_method(
hvac/v1/__init__.py:2711:    @utils.deprecated_method(
hvac/v1/__init__.py:2721:    @utils.deprecated_method(
hvac/v1/__init__.py:2730:    @utils.deprecated_method(
hvac/v1/__init__.py:2740:    @utils.deprecated_method(
hvac/v1/__init__.py:2749:    @utils.deprecated_method(
hvac/v1/__init__.py:2758:    @utils.deprecated_method(
hvac/v1/__init__.py:2765:    @utils.deprecated_method(
hvac/v1/__init__.py:2779:    @utils.deprecated_method(
hvac/v1/__init__.py:2786:    @utils.deprecated_method(
hvac/v1/__init__.py:2796:    @utils.deprecated_method(
hvac/v1/__init__.py:2806:    @utils.deprecated_method(
hvac/v1/__init__.py:2813:    @utils.deprecated_method(
hvac/v1/__init__.py:2820:    @utils.deprecated_method(
hvac/v1/__init__.py:2831:    @utils.deprecated_method(
hvac/v1/__init__.py:2843:    @utils.deprecated_method(
hvac/v1/__init__.py:2853:    @utils.deprecated_method(
hvac/v1/__init__.py:2860:    @utils.deprecated_method(
hvac/v1/__init__.py:2867:    @utils.deprecated_method(
hvac/v1/__init__.py:2887:    @utils.deprecated_method(
hvac/v1/__init__.py:2916:    @utils.deprecated_method(
hvac/v1/__init__.py:2927:    @utils.deprecated_method(
hvac/v1/__init__.py:2936:    @utils.deprecated_method(
hvac/v1/__init__.py:2943:    @utils.deprecated_method(
hvac/v1/__init__.py:2957:    @utils.deprecated_method(
hvac/v1/__init__.py:2966:    @utils.deprecated_method(
hvac/v1/__init__.py:2976:    @utils.deprecated_method(
hvac/v1/__init__.py:2983:    @utils.deprecated_method(
hvac/v1/__init__.py:2990:    @utils.deprecated_method(
hvac/v1/__init__.py:2997:    @utils.deprecated_method(
hvac/v1/__init__.py:3004:    @utils.deprecated_method(
hvac/v1/__init__.py:3011:    @utils.deprecated_method(
hvac/v1/__init__.py:3018:    @utils.deprecated_method(
hvac/v1/__init__.py:3025:    @utils.deprecated_method(
hvac/v1/__init__.py:3033:    @utils.deprecated_method(
hvac/v1/__init__.py:3040:    @utils.deprecated_method(
hvac/v1/__init__.py:3047:    @utils.deprecated_method(
```

### PR DESCRIPTION
**Pull request by colin-pm**
All methods marked as deprecated for the 1.0.0 release have been removed. All references to the deprecated methods in the documentation have been updated. Additionally, all testing has been updated to not use the deprecated methods.

This PR intendeds to close #866.

### CODE REVIEW

**Comment by colin-pm** [comment]
Whoops, forgot to lint, will do that later today and update this change.

**Comment by briantist** [comment]
I would like to bring this one up again:
- https://github.com/hvac/hvac/issues/758

Strangely it seems that the `create_token` method is not removed in this PR?

But more generally, have we been able to identify that all of the methods being removed do have replacements?

File: hvac/v1/__init__.py [hunk 1]
```diff
@@ -298,120 +295,6 @@ def get_policy(self, name, parse=False):
...
-        :return:
-        :rtype:
-        """
-        self._adapter.put("/v1/auth/token/revoke-self")
-
-    @utils.deprecated_method(
-        to_be_removed_in_version="1.0.0",
-        new_method=api.auth_methods.Token.create,
-    )
-    def create_token(
```
Comment (colin-pm) [inline hvac/v1/__init__.py:317]: @briantist create_token is removed here in this PR

**Comment by colin-pm** [comment]
> I would like to bring this one up again:
> 
> * [Does auth.token.create support orphan tokens? #758](https://github.com/hvac/hvac/issues/758)
> 
> Strangely it seems that the `create_token` method is not removed in this PR?

I left a comment where `create_token` was removed in this PR. I read through #758 and the Vault docs and see the issue. Since the behavior seems different between the create token and create orphan token API endpoints, I think that it would be okay to implement a `create_orphan_token` method to hit the other endpoint.

> 
> But more generally, have we been able to identify that all of the methods being removed do have replacements?

I inspected each method that I removed to ensure there was a replacement, and if there were any references to the deprecated method, modified the reference to use the replacement.

The only methods that did not have replacements were the AppID methods. I'm not sure it's worth creating replacements for those, since Vault has indicated that AppID is deprecated in favor of AppRole as of Vault 0.6.1. I would think that if anyone still needs the AppRole method, they can either use an older version of HVAC or contribute an engine to provide support. I don't think it should be a blocker for the 1.0.0 HVAC release.

**Comment by colin-pm** [comment]
I just created #869 to address #758.

**Review by briantist — APPROVED** [review]
This looks good to me, I would just like to see #869 merged before this one

**Pull request merged**